src/test/java/org/opensearch/knn/plugin/script/KNNScriptScoringIT.java

```diff
@@ -683,4 +685,110 @@ public void testKNNInnerProdScriptScore() throws Exception {
+
+        Float[] f2 = { 2.0f, 2.0f };
+        addKnnDoc(INDEX_NAME, "2", FIELD_NAME, f2);
+
+        Float[] f3 = { 4.0f, 4.0f };
+        addKnnDoc(INDEX_NAME, "3", FIELD_NAME, f3);
+
+        Float[] f4 = { 3.0f, 3.0f };
+        addKnnDoc(INDEX_NAME, "4", FIELD_NAME, f4);
+
+        /**
+         * Construct Search Request
+         */
+        QueryBuilder qb = new MatchAllQueryBuilder();
+        Map<String, Object> scriptParams = new HashMap<>();
+        /*
+         *   params": {
+         *       "field": "my_dense_vector",
+         *       "vector": [2.0, 2.0]
+         *      }
+         */
+        float[] queryVector = { 1.0f, 1.0f };
+        scriptParams.put("field", FIELD_NAME);
+        scriptParams.put("query_value", queryVector);
+        scriptParams.put("space_type", SpaceType.L2.getValue());
+        Map<String, Object> searchParams = new HashMap<>();
+        searchParams.put("request_cache", true);
+
+        // first request with request cache enabled
+        Request firstScriptQueryRequest = constructKNNScriptQueryRequest(INDEX_NAME, qb, scriptParams, 4, searchParams);
+        Response firstScriptQueryResponse = client().performRequest(firstScriptQueryRequest);
+        assertEquals(
+            firstScriptQueryRequest.getEndpoint() + ": failed",
+            RestStatus.OK,
+            RestStatus.fromCode(firstScriptQueryResponse.getStatusLine().getStatusCode())
+        );
+
+        List<KNNResult> results = parseSearchResponse(EntityUtils.toString(firstScriptQueryResponse.getEntity()), FIELD_NAME);
+        List<String> expectedDocids = Arrays.asList("2", "4", "3", "1");
+
+        List<String> actualDocids = new ArrayList<>();
+        for (KNNResult result : results) {
+            actualDocids.add(result.getDocId());
+        }
+
+        assertEquals(4, results.size());
+        assertEquals(expectedDocids, actualDocids);
+
+        // assert that the request cache was hit missed at first request
+        Request firstStatsRequest = new Request("GET", "/" + INDEX_NAME + "/_stats");
+        Response firstStatsResponse = client().performRequest(firstStatsRequest);
+        assertEquals(
+            firstStatsRequest.getEndpoint() + ": failed",
+            RestStatus.OK,
+            RestStatus.fromCode(firstStatsResponse.getStatusLine().getStatusCode())
+        );
+        String firstStatsResponseBody = EntityUtils.toString(firstStatsResponse.getEntity());
+        Map<String, Object> firstQueryCacheMap = Optional.ofNullable(
+            createParser(MediaTypeRegistry.getDefaultMediaType().xContent(), firstStatsResponseBody).map()
+        )
+            .map(r -> (Map<String, Object>) r.get("indices"))
+            .map(i -> (Map<String, Object>) i.get(INDEX_NAME))
+            .map(ind -> (Map<String, Object>) ind.get("total"))
+            .map(t -> (Map<String, Object>) t.get("request_cache"))
+            .orElseThrow(() -> new IllegalStateException("Query Cache Map not found"));
```

Comment:
Why cant we access this directly? IRC, index stats always provides you this as response.

Reply:
Yes, this is the index stats API response.